analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -261,6 +282,39 @@ private static void VerifyBuildErrors(ImmutableArray<Diagnostic> diagnostics, Co
 
         private static string IssueType(bool isPrimary) => isPrimary ? "primary" : "secondary";
 
+        internal class File
+        {
+            private string FileName { get; }
+            private SourceText SourceText { get; }
+
+            public File(string fileName)
+            {
+                FileName = fileName;
+                SourceText = SourceText.From(System.IO.File.ReadAllText(fileName));
+            }
+
+            public File(SyntaxTree syntaxTree)
+            {
+                FileName = syntaxTree.FilePath;
+                SourceText = syntaxTree.GetText();
+            }
+
+            public FileNameIssueLocations ToIssueLocations() =>
+                new (FileName, IssueLocationCollector.GetExpectedIssueLocations(SourceText.Lines));
+        }
+
+        internal class FileNameIssueLocations
```

Comment:
For simplicity, this class is friend with the one above :) 
```suggestion
        internal class FileIssueLocations
```